resources/javascripts/script.js

```diff
@@ -0,0 +1,145 @@
+/*-------------------------- slider functonality --------------------------*/
+var slideIndex = 1;
+var showSlides = (n) => {
+    var i;
+    var slides = document.getElementsByClassName("mySlides");
+    var dots = document.getElementsByClassName("dot");
+    if (n > slides.length) { slideIndex = 1 }
+    if (n < 1) { slideIndex = slides.length }
+    for (i = 0; i < slides.length; i++) {
+        slides[i].classList.add("mySlidesHide");
+        slides[i].classList.remove("mySlidesShow");
+    }
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+    slides[slideIndex - 1].classList.add("mySlidesShow");
+    slides[slideIndex - 1].classList.remove("mySlidesHide");
+    dots[slideIndex - 1].className += " active";
+};
+
+var plusSlides = (n) => {
+    showSlides(slideIndex += n);
+};
+
+var currentSlide = (n) => {
+    showSlides(slideIndex = n);
+};
+
+if (document.location.pathname == "/") {
+    showSlides(slideIndex);
+    document.getElementById("prev").addEventListener("click", function() {
+        plusSlides(-1);
+    });
+
+    document.getElementById("next").addEventListener("click", function() {
+        plusSlides(1);
+    });
+
+    document.getElementById("dot").addEventListener("click", function() {
+        currentSlide(this.getAttribute('data-attr'));
+    });
+
+    setInterval(function() { plusSlides(1); }, 1000);
+};
+
+/*-------------------------- Cart functonality --------------------------*/
+var allitemInCart = () => {
+    var url = window.location.origin + "/cart/allitem";
+    let xmlHttpReq = new XMLHttpRequest();
+    xmlHttpReq.open("GET", url, true);
+    xmlHttpReq.onload = () => {
+        if (xmlHttpReq.status == 200) {
+            let data = JSON.parse(xmlHttpReq.responseText);
+            let totalCheckoutAmt = document.getElementById("total-checkout-amt");
+            document.getElementById("item-count").innerHTML = data.itemCounter;
+            let totalCheckoutPrice = 0;
+            data.productInCart.forEach(element => {
+                totalCheckoutPrice = totalCheckoutPrice + element.totalPrice;
+            });
+            totalCheckoutAmt.innerHTML = totalCheckoutPrice;
+        }
+    };
+    xmlHttpReq.send();
+}
+
+var buyItem = (id, operation) => {
+    var url = window.location.origin + '/cart/' + id + '/' + operation;
+    let xmlHttpReq = new XMLHttpRequest();
+    xmlHttpReq.open("GET", url, true);
+    xmlHttpReq.onload = () => {

```

Comment:
have a common/reusable utility function for making XHR calls and call them at appropriate places where required